api.py

```diff
@@ -44,83 +44,145 @@ async def status():
 
 @app.post("/validate")
 async def validate_schema_request_body(payload=Body(None)):
-    logger.info("Validating schema")
+    logger.info("Schema validation request received")
     return await validate_schema(payload)
 
 
 @app.get("/validate")
 async def validate_schema_from_url(url=None):
+    logger.debug("Attempting to validate schema from URL...", url=url)
     if url:
-        logger.info("Validating schema from URL", url=url)
         parsed_url = urlparse(url)
         domain = parsed_url.netloc
         if not is_domain_allowed(parsed_url, domain):
+            logger.warning(
+                "Schema validation request rejected - URL not allowed (this could be related to domain, base_url or repo_owner - see debug logs for values)",
+                url=url,
+            )
             return Response(
                 status_code=400,
                 content=f"URL domain [{parsed_url.hostname}] is not allowed",
             )
-
+        logger.info("Schema validation request accepted - URL allowed", url=url)
         try:
+            # Opens the URL and validates the schema
             with request.urlopen(parsed_url.geturl()) as opened_url:
                 return await validate_schema(data=opened_url.read().decode())
         except error.URLError:
+            logger.warning(
+                "Could not load schema from allowed domain - URL not found", url=url
+            )
             return Response(
-                status_code=404, content=f"Could not load schema at URL [{url}]"
+                status_code=404,
+                content=f"Could not load schema from allowed domain - URL not found [{url}]",
             )
 
 
 async def validate_schema(data):
-    json_to_validate = None
+    logger.debug("Attempting to validate schema from JSON data...")
     if data:
-        if isinstance(data, str):
-            try:
-                json_to_validate = json.loads(data)
-            except JSONDecodeError:
-                logger.info("Could not parse JSON", status=400)
-                return Response(status_code=400, content="Could not parse JSON")
-        elif isinstance(data, dict):
+        # Sets `json_to_validate` to the data received if it is a dictionary as data does not require processing
+        if isinstance(data, dict):
+            logger.info("JSON data received as dictionary - processing not required")
```

Comment:
I'm running it locally now and the `processing not required` could be a little less ambiguous and slightly misleading (too broad term), can we change it to?:
```suggestion
            logger.info("JSON data received as dictionary - decoding and parsing not required")
```
so it's like:
<img width="982" height="739" alt="Untitled 6" src="https://github.com/user-attachments/assets/76372645-484b-467b-8cfb-84b6e10b92de" />